diff --git a/org/libsdl/app/SDLActivity.java b/org/libsdl/app/SDLActivity.java
index 974044f..3257e77 100644
--- a/org/libsdl/app/SDLActivity.java
+++ b/org/libsdl/app/SDLActivity.java
@@ -222,6 +222,10 @@ public class SDLActivity extends Activity implements View.OnSystemUiVisibilityCh
     // This is what SDL runs in. It invokes SDL_main(), eventually
     protected static Thread mSDLThread;
 
+    // Set by PythonActivity.changeKeyboard, which Kivy calls with TextInput.input_type.
+    public static int keyboardInputType = InputType.TYPE_CLASS_TEXT |
+                                          InputType.TYPE_TEXT_FLAG_MULTI_LINE;
+
     protected static SDLGenericMotionListener_API12 getMotionListener() {
         if (mMotionListener == null) {
             if (Build.VERSION.SDK_INT >= 26 /* Android 8.0 (O) */) {
@@ -1941,8 +1945,7 @@ class DummyEdit extends View implements View.OnKeyListener {
     public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
         ic = new SDLInputConnection(this, true);
 
-        outAttrs.inputType = InputType.TYPE_CLASS_TEXT |
-                             InputType.TYPE_TEXT_FLAG_MULTI_LINE;
+        outAttrs.inputType = SDLActivity.keyboardInputType;
         outAttrs.imeOptions = EditorInfo.IME_FLAG_NO_EXTRACT_UI |
                               EditorInfo.IME_FLAG_NO_FULLSCREEN /* API 11 */;
 
